crates/viewer/re_dataframe_ui/src/datafusion_adapter.rs

```diff
@@ -61,6 +61,19 @@ impl From<&TableBlueprint> for DataFusionQueryData {
     }
 }
 
+/// Result of the async datafusion query process.
+#[derive(Debug, Clone)]
+pub struct DataFusionQueryResult {
+    /// The resulting [`SorbetBatch`]s.
+    pub record_batches: Vec<RecordBatch>,
+
+    /// The schema of the record batches (useful if none is returned).
+    pub schema: SchemaRef,
+
+    /// The migrated schema of the record batches.
+    pub sorbet_schema: re_sorbet::SorbetSchema,
```

Comment:
Migrating just the schema seems like a bug waiting to happen. What if we want to join columns in the migrator, for instance?

is the problem is the call to `make_all_data_columns_list_arrays`? If so, just comment it out (I don't think anyone is actually relying on it), or only call it for `BatchType::Chunk`